file/nifile.yml

```diff
@@ -0,0 +1,816 @@
+        orderByDescending value must be a boolean string. The elements in the
+        list are sorted ascending if false and descending if true.
+      operationId: ListAvailableFiles_GET
+      x-ni-operation: listFiles
+      x-ni-privilege: ViewMetadata
+      x-ni-request-all-privileges: true
+      parameters:
+        - in: query
+          name: skip
+          description: How many files to skip in the result when paging
+          type: integer
+          x-ni-data-type: U32
+          default: 0
+        - in: query
+          name: take
+          description: >-
+            How many files to return in the result, or 0 to use a default
+            defined by the service
+          type: integer
+          x-ni-data-type: U32
+          default: 0
+          maximum: 10000
+        - in: query
+          name: orderBy
+          description: The name of the metadata key to sort by
+          type: string
+          enum:
+            - created
+            - id
+            - size
+        - in: query
+          name: orderByDescending
+          description: Whether to sort descending instead of ascending
+          type: boolean
+          default: false
+        - in: query
+          name: id
+          description: Comma-separated list of file IDs to restrict to in the results
+          type: string
+          x-example: '5afb2ce3741fe11d88838cc9,4afb2ce3741fe11d88838cc9'
+      responses:
+        200:
+          $ref: '#/responses/Query'
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/service-groups/Default/files/{id}:
+    parameters:
+      - $ref: '#/parameters/id'
+    delete:
+      tags: [files]
+      summary: Delete file
+      description: Deletes the file indicated by the resource ID.
+      operationId: Delete
+      x-ni-operation: deleteFiles
+      x-ni-privilege: ModifyFiles
+      responses:
+        204:
+          description: No content
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/service-groups/Default/files/{id}/data:
+    parameters:
+      - $ref: '#/parameters/id'
+    get:
+      tags: [files]
+      summary: Download file
+      description: >-
+        Downloads a file from the Skyline File service. The file is returned in
+        a single HTTP response. The file's name metadata property is appended
+        to the response header so that the file is downloaded with the
+        appropriate file name.When the the inline query parameter is not
+        specified or is set to false, the Content-Disposition header is set to
+        'attachment' and the MIME type is set to binary so that the browser
+        handles the response as a file download.  When the inline query
+        parameter is set to true, the Content-Disposition header and MIME type
+        remain unset, and the file contents must be handled by the caller.
```

Comment:
Returns a file from the SystemLink [I think] File service in an HTTP response.

I think the next sentence (about retaining the name) goes without saying, and the sentences that follow can be moved to parameter descriptions.